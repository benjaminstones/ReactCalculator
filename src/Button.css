
   
.calc-button {
    display: inline-flex;
    width: 25%;
    height: 100%;
    flex: 1 0 auto;
    border-radius: 50px;
    margin: 0;
  }
  
  .calc-button.wide {
    width: 50%;
  }
  .calc-button.tall {
    height: 200%;
  }
  
  .calc-button button {
    background-color: #e0e0e0;
    font-size: 1.5rem;
    margin: 5px 2.5px 2.5px 5px;
    flex: 1 0 auto;
    padding: 0;
    border-radius: 8px;
  }
  
 