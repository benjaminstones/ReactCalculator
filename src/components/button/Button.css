  .calc-button {
    display: inline-flex;
    width: 25%;
    height: 100%;
    margin: 0;
  }
  
  .calc-button button {
    background-color: #e0e0e0;
    font-size: 1.5rem;
    margin: 5px 2.5px 2.5px 5px;
    flex: 1 0 auto;
    border-radius: 8px;
  }

  .highlighted-button button{
    background-color: #D0E6F4;
    font-size: 1.5rem;
    margin: 5px 2.5px 2.5px 5px;
    flex: 1 0 auto;
    border-radius: 8px;
  }
 