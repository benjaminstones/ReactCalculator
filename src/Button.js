import './Button.css'

function Button(props) {
    const handleClick = () => {
        props.clickHandler(props.btnName, props.btnType);
      };

    return (
      <div className='calc-button'>
        <button onClick={handleClick}>{props.btnName}</button>
      </div>
    );
  }
  
  export default Button;