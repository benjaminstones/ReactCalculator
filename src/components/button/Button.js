import { useEffect, useState } from 'react';
import { OPERATOR_CHAR, UTIL } from '../../Constants';
import './Button.css';

function Button(props) {
  const handleClick = () => {
    props.clickHandler(props.btnName, props.btnType);
  };
  const [isHighlighted, setIsHighlighted] = useState(false);

  useEffect(() => {
    if (props.btnName === 'AC' || props.btnType === UTIL || props.btnType === OPERATOR_CHAR) {
      console.log(true);
      setIsHighlighted(true);
    }
  });

  return (
    <div className={isHighlighted ? 'calc-button highlighted-button' : 'calc-button'}>
      <button onClick={handleClick}>{props.btnName}</button>
    </div>
  );
}

export default Button;