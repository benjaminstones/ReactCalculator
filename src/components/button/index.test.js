import React from 'react';
import { shallow } from 'enzyme';
import '../../setupTests'
import Button from './index.js';
import HIGHLIGHTED from '../../constants/index.js';

let wrapped = shallow(<Button btnName={'AC'} btnTags={{ HIGHLIGHTED }} />);
describe('Button', () => {
	it('should render the Button Component correctly', () => {
		expect(wrapped).toMatchSnapshot();
	});
});