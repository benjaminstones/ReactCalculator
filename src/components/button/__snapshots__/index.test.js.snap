// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the Button Component correctly 1`] = `
<styled.div>
  <styled.button
    onClick={[Function]}
  >
    AC
  </styled.button>
</styled.div>
`;
